topostats/run_topostats.py

```diff
@@ -37,7 +48,47 @@
 # pylint: disable=too-many-nested-blocks
 
 
-def run_topostats(args: None = None) -> None:  # noqa: C901
+def handle_config_file(args: argparse.Namespace | None) -> dict:
+    """
+    Handle the config file argument.
+
+    Parameters
+    ----------
+    args : argparse.Namespace
+        Command line arguments passed into TopoStats.
+
+    Returns
+    -------
+    dict
+        The configuration dictionary.
+    """
+    default_config_raw = (resources.files(__package__) / "default_config.yaml").read_text()
+    default_config = yaml.safe_load(default_config_raw)
+    if args is not None:
+        config_file_arg: str | None = args.config_file
+        if config_file_arg is not None:
+            config = read_yaml(config_file_arg)
+            # Merge the loaded config with the default config to fill in any defaults that are missing
+            # Make sure to prioritise the loaded config, so it overrides the default
+            config = merge_mappings(map1=default_config, map2=config)
+        else:
+            # If no config file is provided, use the default config
+            config = default_config
+    else:
+        # If no args are provided, use the default config
+        config = default_config
+
+    # Override the config with command line arguments passed in, eg --output_dir ./output/
+    if args is not None:
+        config = update_config(config, args)
```

Comment:
Would it be worthwhile replacing `update_config()` in favour of the typed `merge_mappings()`? They are doing the same thing, although `update_config()` doesn't use the `MutableMapping` type and also converts path strings to `Path()`.

Reply:
I'm so sorry I didn't realise I had re-written your function. I assumed update_config did something different!

Don't want to tread on toes, so happy to remove `merge_mappings` in favour of `update_config` if you like!

Reply:
Don't worry its only code and I'm not possessive about it.

I don't mind which remains bit glad that we converged on the same solution!

Go with whichever you think is best, currently I think that is `merge_mappings()`, stronger type-hints and clearer docstring.

Reply:
Tried replacing it, but they do slightly different things, in that `update_config` takes `argparse.Namespace` objects, while `merge_mappings` takes only `MutableMappings`, attempting to replace causes errors with iteration. I tried to fix this by using `vars(args)` on the input but it still fails, so if it's okay with you, I'd suggest keeping both for now?

Reply:
Ok, keep both but write up an issue to go through and reconcile the two in the future. On the surface they look very similar so it should be possible to write one function.